javers-persistence-sql/src/main/java/org/javers/repository/sql/finders/SnapshotQuery.java

```diff
@@ -9,24 +9,53 @@
 import org.javers.repository.sql.session.Parameter;
 import org.javers.repository.sql.session.SelectBuilder;
 import org.javers.repository.sql.session.Session;
+import org.polyjdbc.core.dialect.Dialect;
+import org.polyjdbc.core.dialect.DialectRegistry;
 
+import java.sql.Clob;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.stream.Collectors;
 
-import static org.javers.repository.sql.schema.FixedSchemaFactory.*;
-import static org.javers.repository.sql.session.Parameter.*;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.COMMIT_AUTHOR;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.COMMIT_COMMIT_DATE;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.COMMIT_COMMIT_DATE_INSTANT;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.COMMIT_COMMIT_ID;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.COMMIT_PK;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.COMMIT_PROPERTY_COMMIT_FK;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.COMMIT_PROPERTY_NAME;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.COMMIT_PROPERTY_VALUE;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.GLOBAL_ID_FRAGMENT;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.GLOBAL_ID_LOCAL_ID;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.GLOBAL_ID_OWNER_ID_FK;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.GLOBAL_ID_PK;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.GLOBAL_ID_TYPE_NAME;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.SNAPSHOT_CHANGED;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.SNAPSHOT_COMMIT_FK;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.SNAPSHOT_GLOBAL_ID_FK;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.SNAPSHOT_MANAGED_TYPE;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.SNAPSHOT_PK;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.SNAPSHOT_STATE;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.SNAPSHOT_TYPE;
+import static org.javers.repository.sql.schema.FixedSchemaFactory.SNAPSHOT_VERSION;
+import static org.javers.repository.sql.session.Parameter.bigDecimalParam;
+import static org.javers.repository.sql.session.Parameter.instantParam;
+import static org.javers.repository.sql.session.Parameter.localDateTimeParam;
+import static org.javers.repository.sql.session.Parameter.longParam;
+import static org.javers.repository.sql.session.Parameter.stringParam;
 
 class SnapshotQuery {
```

Comment:
If Dialect is needed in SnapshotQuery, maybe you can get it from selectBuilder.session.dialect?

Reply:
I have pushed a new version. I have used session.dialect, but the classes Session, Dialect and Dialects are more public now and it is of course a drawback. I can not say whether it is better comparing with the previous version with additional injections.

Reply:
ok, I will take a look